### PR TITLE
Assignment Updates: SectionAssigner on unit overview

### PR DESCRIPTION
**Pull request by Erin007**
Similar to #31315. This PR uses the `SectionAssigner` component on the unit  Overview page with real data behind the "assignmentUpdates"  experiment flag. For the selected section, if the current user has assigned or been assigned the unit there will be a green checkmark and the Assigned button.
Screen Shot 2019-10-17 at 1 48 16 PM

- [spec](https://docs.google.com/document/d/1vJySqyKozUxknkhY1KyQXWzbV-0c1vdnpKvRsAsVrdI/edit)
- [jira LP-706](https://codedotorg.atlassian.net/browse/LP-706) partial

Follow up work:  
There is some functionality in the current SectionSelector not yet implemented in SectionAssigner. 

- [ ] update query params to reflect the id of the selected section

### CODE REVIEW

File: apps/src/code-studio/components/progress/ScriptOverviewTopRow.jsx [hunk 1]
```diff
@@ -105,6 +112,12 @@ export default class ScriptOverviewTopRow extends React.Component {
               assignmentName={scriptTitle}
             />
           )}
+        {experiments.isEnabled(experiments.ASSIGNMENT_UPDATES) && (
+          <SectionAssigner
+            sections={sections}
+            initialSelectedSectionId={selectedSectionId}
```
Comment (davidsbailey) [inline apps/src/code-studio/components/progress/ScriptOverviewTopRow.jsx:119]: please hide the old assign button when the experiment is enabled

File: dashboard/app/views/scripts/show.html.haml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 -# This variable will get any data we need to pass along to scripts/show.js
 - script_data = @script.summarize(true, current_user)
-- additional_script_data = {course_name: @script.course&.name, show_redirect_warning: @show_redirect_warning, redirect_script_url: @redirect_script_url, section: @section, user_type: current_user&.user_type, user_id: current_user&.id, is_verified_teacher: current_user&.authorized_teacher?, locale: Script.locale_english_name_map[request.locale]}
+- sections = @current_user.try{|u| u.sections.where(hidden: false).select(:id, :name, :script_id)}
+- sections = sections.map { |section| section.attributes.merge!({"isAssigned" => section[:script_id] == @script.id})}
```
Comment (davidsbailey) [inline dashboard/app/views/scripts/show.html.haml]: apologies for not commenting on this earlier, but the best practice is for ruby code to go in the controller method: https://github.com/code-dot-org/code-dot-org/blob/0b2856ab5d99d3fa1e6e1ecc7277957f9ec18358/dashboard/app/controllers/scripts_controller.rb#L12

one benefit is that our linting rules will run on it, catching potential errors. 

In this case, please at least move the definition of `sections` to the controller method, name it `@sections` and then access it below as `@sections`.

ideally, all of `scriptOverviewData` would move to the controller method but it's ok if you don't want to tackle that now.

**Review by davidsbailey — APPROVED** [review]

File: apps/src/code-studio/components/progress/ScriptOverviewTopRow.jsx (hunk 1 above)
Comment (Erin007) [reply apps/src/code-studio/components/progress/ScriptOverviewTopRow.jsx:119]: Gah! Sorry I should've caught that after the last reminder. 
Screen Shot 2019-10-18 at 3 31 06 PM

File: dashboard/app/views/scripts/show.html.haml (hunk 2 above)
Comment (Erin007) [reply dashboard/app/views/scripts/show.html.haml]: done. updated here and for courses.

**Review by davidsbailey — APPROVED** [review]
Looks great, thanks for the follow-up @Erin007 !

**Review by dmcavoy — APPROVED** [review]
Looks good Erin!

**Pull request merged**